mido/midifiles/midifiles.py

```diff
@@ -294,13 +295,9 @@ def get_seconds_per_tick(tempo, ticks_per_beat):
 
 
 class MidiFile(object):
-    def __init__(self, filename=None, file=None,
-                 type=1, ticks_per_beat=DEFAULT_TICKS_PER_BEAT,
-                 charset='latin1',
-                 debug=False,
-                 clip=False,
-                 tracks=None
-                 ):
+    def __init__(self, filename: str = None, file: IO[bytes] = None, type: int = 1,
```

Comment:
I thought that it has to be `filename: Optional[str] = None` and the like.
Does the current state pass a `mypy` run?